x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/ml_inference_logic.ts

```diff
@@ -137,29 +166,57 @@ export const MLInferenceLogic = kea<
       } = values;
 
       actions.makeCreatePipelineRequest({
-        indexName,
-        pipelineName: configuration.pipelineName,
-        modelId: configuration.modelID,
-        sourceField: configuration.sourceField,
         destinationField:
           configuration.destinationField.trim().length > 0
             ? configuration.destinationField
             : undefined,
+        indexName,
+        modelId: configuration.modelID,
+        pipelineName: configuration.pipelineName,
+        sourceField: configuration.sourceField,
       });
     },
     makeCreatePipelineRequest: () => actions.setCreateErrors([]),
     setIndexName: ({ indexName }) => {
       actions.makeMLModelsRequest(undefined);
       actions.makeMappingRequest({ indexName });
     },
+    simulatePipeline: () => {
+      if (values.mlInferencePipeline) {
+        actions.setSimulatePipelineErrors([]);
+        actions.makeSimulatePipelineRequest({
+          docs: values.addInferencePipelineModal.simulateBody,
+          indexName: values.addInferencePipelineModal.indexName,
+          pipeline: values.mlInferencePipeline,
+        });
+      }
+    },
   }),
+  path: ['enterprise_search', 'content', 'inference_pipelines'],
   reducers: {
     addInferencePipelineModal: [
       {
         configuration: {
           ...EMPTY_PIPELINE_CONFIGURATION,
         },
         indexName: '',
+        simulateBody: `
```

Comment:
question: is this actual commitable code we'll show users, or just here to test things?